plenum/test/demotion_promotion/test_demote_promote.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from plenum.test.helper import checkViewNoForNodes, waitForViewChange, sdk_send_random_and_check
+from plenum.test.node_catchup.helper import ensure_all_nodes_have_same_data
+from plenum.test.node_request.helper import sdk_ensure_pool_functional
+from plenum.test.pool_transactions.helper import demote_node, promote_node
+from plenum.test.test_node import ensureElectionsDone
+from plenum.test.view_change.helper import restart_node
+
+nodeCount = 6
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    old_b_size = tconf.Max3PCBatchSize
+    tconf.Max3PCBatchSize = 1
+
+    yield tconf
+    tconf.Max3PCBatchSize = old_b_size
+
+
+def test_demote_promote_restart_after_promotion(txnPoolNodeSet,
```

Comment:
It would be interesting to have the same test where F is changed as a result of demotion/promotion (so nodeCount=7)

Reply:
Ok, i will add this test.